Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3026,6 +3145,82 @@ public void run() {
 		});
 	}
 
+	/*
+    * Checks if the BranchActivityLifeCycleObserver is initialised
+    * @return true if BranchActivityLifeCycleObserver initialised else false
+    */
+	public boolean isActivityObserverInitialised(){
+		return activityLifeCycleObserver_ != null;
+	}
+
+	@TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+	private void setActivityLifeCycleObserver(Application application){
+		application.registerActivityLifecycleCallbacks(new BranchActivityLifeCycleObserver());
+	}
+
+	/**
+	 * <p>Class that observes activity life cycle events and determines when to start and stop
+	 * session.</p>
+	 */
+	@TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+	private class BranchActivityLifeCycleObserver implements Application.ActivityLifecycleCallbacks{
+		int m_ActivityCnt = 0; //Keep the count of live  activities
+
+		public BranchActivityLifeCycleObserver(){
+			activityLifeCycleObserver_ = this;
+		}
+
+		@Override
+		public void onActivityCreated(Activity activity, Bundle bundle) {}
+
+		@Override
+		public void onActivityStarted(Activity activity) {
+			if(m_ActivityCnt < 1){ //Check if this is the first Activity.If so start a session
+				onSessionStarted();
```

Comment:
onSessionStarted isn't called anywhere else, can we just call `initSession()` here instead and remove `private void onSessionStarted()` (same for onSessionEnded())?